transcendence-app/src/shared/avatar/avatar.service.ts

```diff
@@ -40,7 +44,17 @@ export class AvatarService {
       fileTypeResult && AVATAR_MIMETYPE_WHITELIST.includes(fileTypeResult.mime);
     if (!isValid) {
       this.localFileService.deleteFileData(path);
+      const allowedTypes = AVATAR_MIMETYPE_WHITELIST.join(', ');
+      throw new UnprocessableEntityException(
+        `Validation failed (allowed types are ${allowedTypes})`,
+      );
```

Comment:
Nice but isn't this evaluated in the interceptor?

Reply:
The interceptor checks the mime type, and this function uses the `file-type` library to check the binary format.

For example, someone could rename a `.pdf` file to `.jpeg`, and the mime type is `image/jpeg`. In this case, the mime type, according to the binary format, is `application/pdf`